src/main/java/com/hms_networks/americas/sc/extensions/logging/Logger.java

```diff
@@ -491,4 +574,32 @@ public static void DUMP_LOG_QUEUE() {
       LOG(LOG_LEVEL_CRITICAL, logQueue.poll());
     }
   }
+
+  /**
+   * Sets whether a specific log has been outputted. This is used to prevent spamming the log with
+   * the same message.
+   *
+   * @param logKey key for the log event
+   * @param logOutputted boolean representing whether the log has been output (true) or not (false)
+   * @since 1.15.9
+   */
+  public static void setLoggedOnce(String logKey, boolean logOutputted) {
+    if (logOutputted) {
+      logOnceList.add(logKey);
+    } else {
+      logOnceList.remove(logKey);
+    }
+  }
+
+  /**
+   * Gets a boolean indicating whether a specific log has been outputted. This is used to prevent
+   * spamming the log with the same message.
+   *
+   * @param logKey key for the log event
+   * @return {@code true} if the log has been outputted, {@code false} otherwise
+   * @since 1.15.9
+   */
+  public static boolean getLoggedOnce(String logKey) {
+    return logOnceList.contains(logKey);
```

Comment:
contains is O(n) for time
moving to a hash map ( or something similar) would be O(1). The idea is that the key would be the hash.

Reply:
Migrated to a HashSet-based implementation, which in Java is backed by a HashMap.